src/FileUpload/sgds-fileupload.ts

```diff
@@ -0,0 +1,205 @@
+import { html } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import styles from "./sgds-fileupload.scss";
+import SgdsElement from "../utils/sgds-element";
+import { unsafeSVG } from "lit/directives/unsafe-svg.js";
+import genId from "../utils/generateId";
+import { createRef, ref } from "lit/directives/ref.js";
+
+export type ButtonVariant =
+  | "primary"
+  | "secondary"
+  | "success"
+  | "danger"
+  | "warning"
+  | "info"
+  | "light"
+  | "dark"
+  | "link"
+  | "outline-primary"
+  | "outline-secondary"
+  | "outline-success"
+  | "outline-danger"
+  | "outline-warning"
+  | "outline-info"
+  | "outline-light"
+  | "outline-dark";
+
+type SelectedFileType = FileList | {};
+
+@customElement("sgds-fileupload")
+export class SgdsFileUpload extends SgdsElement {
+  /** The button's variant. */
+  @property({ reflect: true }) variant: ButtonVariant = "primary";
+
+  // /** Sets a unique id to the file input, required. */
+  // @property({ type: String }) controlId = "";
+
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  /** Button sizes */
+  @property({ reflect: true }) size: "sm" | "lg";
+
+  @property({ type: String })
+  checkedIcon? = "";
+
+  @property({ type: String })
+  cancelIcon? = "";
+
+  @property({ type: FileList })
+  private fileList: FileList | undefined;
+
+  @property({ type: Array })
+  private selectedFiles: File[] = [];
+
+  constructor() {
+    super();
+    this.addEventListener("sgds-change", () => {
+      this.selectedFiles = Array.from(this.fileList || []);
+      this.dispatchEvent(
```

Comment:
- use this.emit() instead

- the event name should follow the event naming convention starting with sgds-<action-words> , like sgds-files-selected